src/coreclr/nativeaot/Runtime/interoplibinterface_objc.cpp

```diff
@@ -0,0 +1,187 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+#include "common.h"
+#include "CommonTypes.h"
+#include "CommonMacros.h"
+#include "daccess.h"
+#include "PalRedhawkCommon.h"
+#include "PalRedhawk.h"
+#include "rhassert.h"
+#include "slist.h"
+#include "holder.h"
+#include "gcrhinterface.h"
+#include "shash.h"
+#include "RWLock.h"
+#include "rhbinder.h"
+#include "Crst.h"
+#include "RuntimeInstance.h"
+#include "TypeManager.h"
+#include "MethodTable.h"
+#include "ObjectLayout.h"
+#include "event.h"
+#include "varint.h"
+#include "regdisplay.h"
+#include "StackFrameIterator.h"
+#include "thread.h"
+#include "threadstore.h"
+#include "threadstore.inl"
+
+#include "interoplibinterface.h"
+
+#include "MethodTable.inl"
+
+#ifdef FEATURE_OBJCMARSHAL
+
+namespace
+{
+    struct GcBeginEndCallbacks
+    {
+        ObjCMarshalNative::BeginEndCallback m_pCallbackFunction;
+        // May be NULL for the end of the linked list
+        GcBeginEndCallbacks * m_pNext;
+    };
+
+    CrstStatic s_sLock;
+    GcBeginEndCallbacks * m_pBeginEndCallbacks = nullptr;
+
+    void CallBeginEndCallbacks()
+    {
+        GcBeginEndCallbacks * pCallback = m_pBeginEndCallbacks;
+        while (pCallback != nullptr)
+        {
+            ObjCMarshalNative::BeginEndCallback fn = pCallback->m_pCallbackFunction;
+            fn();
+            pCallback = pCallback->m_pNext;
+        }
+    }
+
+    bool TryGetTaggedMemory(_In_ Object * obj, _Out_ void ** tagged)
+    {
+        void* fn = obj->get_EEType()
```

Comment:
Was the intent to use `TryGetCallbackViaClasslib` here too?

Reply:
This `ObjectiveCMarshalTryGetTaggedMemory` class lib callback, unlike the other two, can be called directly. (The other two return a function pointer to managed code that needs to be called to get the real callback).

Anyways, since I was initially confused myself re-reading the code after this comment, I added some more comments and changed the name of the callback-getting function. Hopefully it is less confusing to myself next time I read it.